src/ucp/core/ucp_worker.c

```diff
@@ -333,6 +333,7 @@ static void ucp_worker_iface_disarm(ucp_worker_iface_t *wiface)
     ucs_status_t status;
 
     if (wiface->flags & UCP_WORKER_IFACE_FLAG_ON_ARM_LIST) {
+        ucs_assert(wiface->attr.cap.event_flags & UCT_IFACE_FLAG_EVENT_FD);
```

Comment:
maybe add helper inline function
ucp_worker_iface_get_eventfd() which will also have the assertion

Reply:
done